docs/Talkbuchet.js

```diff
@@ -124,11 +170,99 @@
+ *
+ * Virtual participant mode is used for load testing of clients. From the point
+ * of view of the clients in a call the virtual participants are real
+ * participants (they can be just listeners, but they can publish audio and/or
+ * video), so several virtual participants can be added to a call to verify if
+ * a client running on a specific system will be able to withstand certain
+ * number of participants in a call.
+ *
+ * The reason to use virtual participants instead of real participants is that
+ * virtual participants are either just in the call or publishing media, but
+ * they will not subscribe to any other participant (which does not affect the
+ * other clients, only the HPB server). Due to this they need much less
+ * resources than real participants and therefore the system launching the test
+ * would be able to add a much higher number of virtual participants than of
+ * real participants.
+ *
+ * However, note that each web browser session can execute a single virtual
+ * participant. Due to this it is recommended to use Talkbuchet CLI instead to
+ * easily start several web browser sessions, each one with its own virtual
+ * participant, and control the virtual participants from the CLI.
+ *
+ * HOW TO RUN:
+ * -----------------------------------------------------------------------------
+ * - Set the user and appToken (generate an apptoken at
+ *   index.php/settings/user/security) by calling
+ *   "setCredentials(user, appToken)" in the console. If credentials are not set
+ *   a guest user will be used instead.
+ * - Set the token of the conversation by calling "setToken(token)" in the
+ *   console.
+ * - Start the desired media by calling "startMedia(audio, video)" in the
+ *   console. If not called (or both parameters are false) no media will be
+ *   used.
+ * - Once all the needed parameters are set execute "startVirtualParticipant()"
+ *   in the console.
+ * - To run it again execute "stopVirtualParticipant()" and then
+ *   "startVirtualParticipant()" again in the console; if any parameter
+ *   needs to be changed do it before starting the virtual participant again.
+ *
+ * TRIGGERING ACTIONS BY THE VIRTUAL PARTICIPANT:
+ * -----------------------------------------------------------------------------
+ * In general, for load testing of clients it is enough to start the virtual
+ * participant and that is it. However, for development purposes it can be
+ * useful to simulate certain actions by the virtual participant.
+ *
+ * Currently all the actions are simulated through data channel messages; the
+ * equivalent signaling messages are not sent.
+ *
+ * The nick of the virtual participant can be set with:
+ * sendNickThroughDataChannel(NICK)
+ *
+ * Note that sending the nick is not limited to guests; even if the virtual
+ * participant is a registered user a nick can be sent. Moreover, clients may
+ * not show any name at all for the virtual participant of a registered user
```

Comment:
That sounds like a bug to be reported?
But boils down to the old topic of getting rid of data channels?